FreydCategoriesForCAP/gap/CategoryOfRows.gd

```diff
@@ -53,6 +53,89 @@ DeclareOperation( "\/",
 
+####################################
+##
+#! @Section Random Methods in Category of Rows
+##
+####################################
+
+if false then
+
+#! @Description
+#! The arguments are a category <A>C</A> and a list <A>L</A> of non-negative integers. The output is an object whose
+#! rank is a random element in <A>L</A>.
+#! @Returns an object in a category of rows
+#! @Arguments C, L
+DeclareOperation( RandomObjectByList, [ IsCategoryOfRows, IsList ] );
+
+#! @Description
+#! The arguments are a category of rows <A>C</A> and a non-negative integer <A>n</A>. The output is an object of rank at most <A>n</A>.
+#! @Returns an object in a category of rows
+#! @Arguments C, n
+DeclareOperation( RandomObjectByInteger, [ IsCategoryOfRows, IsInt ] );
+
+#! @Description
+#! The arguments are an object <A>S</A> and a list <A>L</A> of non-negative integers.
+#! The output is a morphism from <A>S</A> to an object <A>R</A> whose rank is randomly chosen from <A>L</A>.
+#! @Returns a morphism in a category of rows
+#! @Arguments S, L
+DeclareOperation( RandomMorphismWithFixedSourceByList, [ IsCategoryOfRowsObject, IsList ] );
+
+#! @Description
+#! The arguments are an object <A>S</A> and a non-negative integer <A>n</A>.
+#! The output is a morphism from <A>S</A> to an object <A>R</A> whose rank is at most <A>n</A>.
+#! @Returns  a morphism in a category of rows
+#! @Arguments S, n
+DeclareOperation( RandomMorphismWithFixedSourceByInteger, [ IsCategoryOfRowsObject, IsInt ] );
+
+#! @Description
+#! The arguments are an object <A>R</A> and a list <A>L</A> of non-negative integers.
+#! The output is a morphism from an object <A>S</A> whose rank is randomly chosen from <A>L</A> to <A>R</A>.
+#! @Returns a morphism in a category of rows
+#! @Arguments R, L
+DeclareOperation( RandomMorphismWithFixedRangeByList, [ IsCategoryOfRowsObject, IsList ] );
+
+#! @Description
+#! The arguments are an object <A>R</A> and a non-negative integer <A>n</A>.
+#! The output is a morphism from an object <A>S</A> whose rank is at most <A>n</A> to <A>R</A>.
+#! @Returns a morphism in a category of rows
+#! @Arguments R, n
+DeclareOperation( RandomMorphismWithFixedRangeByInteger, [ IsCategoryOfRowsObject, IsInt ] );
+
+#! @Description
+#! The arguments are two objects <A>S</A>, <A>R</A> and a list <A>L</A> of integers or elements in the underlying ring of the category.
+#! The output is a morphism from <A>S</A> to <A>R</A> whose matrix is an <A>L</A>-linear combination of random matrices over the underlying ring.
+#! @Returns  a morphism in a category of rows
+#! @Arguments S, R, L
+DeclareOperation( RandomMorphismWithFixedSourceAndRangeByList, [ IsCategoryOfRowsObject, IsCategoryOfRowsObject, IsList ] );
+
+#! @Description
+#! The arguments are two objects <A>S</A>, <A>R</A> and a non-negative integer <A>n</A>.
+#! The output is a morphism from <A>S</A> to <A>R</A> whose matrix is the sum of <A>n</A> random matrices over the underlying ring.
```

Comment:
```suggestion
#! The output is a morphism from <A>S</A> to <A>R</A> whose matrix is the sum of <A>n</A> random matrices over the underlying ring multiplied by 1,…,<A>n<A/>, respectively.
```
Is this desirable? This means that for large `n` the matrix will probably have large entries, except if some entries cancel by chance. Maybe just taking a single random matrix (i.e. ignoring the integer in this case) would be more desirable?

Reply:
Let me know if you have other suggestions.